### PR TITLE
fix: [FFM-10244]: Updating yup validation for Target/Group fix

### PR DESCRIPTION
**Pull request by laura-mannus**
### Summary

Amending the fix that caused existing targeting to not be editable.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by laura-mannus** [comment]
retrigger jest

**Comment by laura-mannus** [comment]
retrigger coverage

**Review by Kira-Jennings — APPROVED** [review]
Looks good!

**Review by crystal-fong — APPROVED** [review]

**Review by Kira-Jennings — APPROVED** [review]

File: src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/hooks/useTargetingRulesFormValidation.tsx [hunk 1]
```diff
@@ -42,20 +42,33 @@ const useTargetingRulesFormValidation = (): UseTargetingRulesFormValidationRetur
...
               }
 
               if ('variations' in (v as VariationPercentageRollout)) {
                 validations.variations = getPercentageRolloutVariationsArrayTest(getString)
               }
 
-              return yup.object(validations)
+              if ('targets' in (v as FormVariationMap) || 'targetGroups' in (v as FormVariationMap)) {
+                return yup.object({
+                  targets: (v as FormVariationMap).targetGroups?.length
```
Comment (crystal-fong) [inline src/modules/75-cf/pages/feature-flags-detail/targeting-rules-tab/hooks/useTargetingRulesFormValidation.tsx:53]: should this be
```suggestion
                  targets: (v as FormVariationMap).target?.length
```

**Review by crystal-fong — APPROVED** [review]

**Review by knagurski — APPROVED** [review]
Noice

**Pull request merged**